### ISSUE TITLE
All rules not working when one value mapping rule get the "null" metric

### ISSUE THREAD

**Issue by Sunnor** [algenty/grafana-flowcharting#65]
Hi Arnaud,
I want to report a bug, would you help to check?

**[Issue]:**   All Text Mapping and Value Mapping rules will be not working when a Value Mapping rule get the "null " metric value.
**[Version]:** 0.4.0
**[Reproduce Steps]:** 
                  1.  Create a panel and add some mapping rules.
                  2.  Panel display normally.
![image](https://user-images.githubusercontent.com/49578115/67737668-1bf25880-fa47-11e9-82ad-6ac6d63adc90.png)
                  3. Due to some database networking issue, grafana may query some metrics with "null" value sometimes.  If this metric is bonding to a value mapping rule, then all rules will be not working. And there will an error **cannot read property "toString" of null**
![image](https://user-images.githubusercontent.com/49578115/67737558-9e2e4d00-fa46-11e9-91b9-41cb0c86ad98.png)
![image](https://user-images.githubusercontent.com/49578115/67737631-ee0d1400-fa46-11e9-9771-223b1bb892e6.png)

**[Desired result]:**  It should only affect the specific rule that have the problem, instead of broking all rules.

**Comment by algenty** [comment]
Hi Sunor, 
Can you give stack trace in console  plz ?
have you try version 0.5.0 ?

BR
Arnaud

**Comment by Sunnor** [comment]
Hi Arnaud,

Do you mean this one?
![image](https://user-images.githubusercontent.com/49578115/67753773-3a207e80-fa70-11e9-9f29-327f8142911b.png)
---------------------------------------------
metrics_panel_ctrl.ts:179 Panel data error: TypeError: Cannot read property 'toString' of null
    at Rule.getFormattedValue (rule_class.js:697)
    at State.setState (state_class.js:92)
    at eval (statesHandler.js:256)
    at Array.forEach (<anonymous>)
    at eval (statesHandler.js:255)
    at Array.forEach (<anonymous>)
    at eval (statesHandler.js:254)
    at Array.forEach (<anonymous>)
    at StateHandler.setStates (statesHandler.js:252)
    at Flowchart.setStates (flowchart_class.js:143)
---------------------------------------------

And I haven't tried 0.5.0 yet. I'll have a try and update the testing result

**Comment by algenty** [comment]
what is the type of your value ? string ? number ? date ?

**Comment by Sunnor** [comment]
1. I can reproduce this issue with version 0.5.0

**Panel data error: TypeError: Cannot read property 'toString' of null
    at Rule.getFormattedValue (rule_class.js:541)
    at State.setState (state_class.js:122)
    at eval (statesHandler.js:159)
    at Array.forEach (<anonymous>)
    at eval (statesHandler.js:158)
    at Array.forEach (<anonymous>)
    at eval (statesHandler.js:157)
    at Array.forEach (<anonymous>)
    at StateHandler.setStates (statesHandler.js:155)
    at Flowchart.setStates (flowchart_class.js:113)**


2. I have number values and string values in this diagram, but the one that leading the issue is a string value with "Value Mappings" rule. 

3. BTW, 0.5.0 looks awesome, great job!

**Comment by algenty** [comment]
I think i fix it during my lunch, can you try it ?
on https://algenty.github.io/flowcharting-repository/ARCHIVES.html
0.6.0 snapshot

**Comment by Sunnor** [comment]
It's good now with your 0.6.0 snapshot.

One or two value mapping rules getting "null" value will not break all rules now. it will show a "-" instead.
![image](https://user-images.githubusercontent.com/49578115/67823948-f07f7480-faff-11e9-9ec6-f1aaa2b181c6.png)

Thanks a lot for quick fix.

Looking forward to having the 0.6.0 version.

**Comment by algenty** [comment]
You're welcome 👍 
I choose '-' when metric is null, i don't know if that is a good choise, you can change that with Value mapping : - => ' ' in rule.
Have you try graph tooltip on your columns Google/facebook ? be carefull histogram test is in progress in 0.6.0, Line Graph works very well for me.

Can i had you screenshot for tweeter and gallery ?

BR
Arnaud

**Comment by Sunnor** [comment]
Hi Arnaud,

1. I tried map "-" to other string like "Error", but it seems not working, it still display "-".

---------
![image](https://user-images.githubusercontent.com/49578115/67846722-aa94d180-fb3c-11e9-8cc9-93f29827eca7.png)


2. Yes, I've tried graph tooltip in both 0.5.0 and 0.6.0, it works fine.  This feature looks awesome.

3.  Sorry, I didn't get your meaning of "tweeter and gallery",  may I know what they are?
